[Lens] Document the "Other" entry in data summaries in Kibana guide (#85575)
elastic/kibana: docs/user/dashboard/dashboard.asciidoc

[role="screenshot"]
image::images/lens_data_info.png[Data summary window]

> *Lens* inspects the top 10 values to build the value distribution: every other value, together with documents with no data for the inspected field, are shown with the "Other" entry in the distribution chart.

Reviewer: This sentence is missing information about what happens for number/date fields: in both cases we do something more than just top 10. For numbers we show both top 10 and histogram. We show the histogram when the top 10 values are less than 10% of all the values.

[role="screenshot"]
image::images/lens_data_info_other.png[Data summary window with Other]

> NOTE: because the distribution is based on single values, the sum of all the entries and "Other" may be above 100%, because entries in array values are counted as distinct values.

Reviewer 2: I did a quick test and while Tim raises a good point in the issue you opened, this is not the case right now:
<img width="308" alt="Screenshot 2020-12-14 at 15 25 17" src="https://user-images.githubusercontent.com/1508364/102092676-9aaf7e00-3e20-11eb-8186-3471626216b1.png">

For this data:
```
POST testindex/_doc
{
  "val": ["A", "B", "C"]
}

POST testindex/_doc
{
  "val": ["D", "E", "F"]
}

POST testindex/_doc
{
  "val": ["G", "H", "I"]
}
```

Right now it seems like this only caused by rounding

Reviewer: Catching up on this discussion: @flash1293 is right that it's only a rounding issue, not an issue with the underlying calculation. The size of "other" is based on the `value_count` of the field minus the sum of all terms we get back, which is correct for array fields and regular fields.

[role="screenshot"]
image::images/lens_data_info.png[Data summary window]

> Depending on the type of data, *Lens* shows a summary that better suits the field analysis: for date types the time distribution, for string types the top 10 values and for numeric types are shown a detailed summary with the top 10 values and a value distribution.

Reviewer: Sentence is looking good, just want to tweak the grammar to use a more active voice instead of passive ("show" instead of "are shown")

```suggestion
*Lens* shows a summary depending on the type of data: date fields show the time distribution, string fields show the top 10 values, and numeric fields show a detailed summary with the top 10 values and a value distribution.
```

[role="screenshot"]
image::images/lens_data_info_documents.png[Data summary analyzed documents]

> *Lens* uses a sample of 5000 documents to perform the fields analysis, therefore the bottom line of the summary hints how many documents have been analyzed to build the distribution.

Reviewer: ```suggestion
*Lens* uses a sample of 5,000 documents to perform the field analysis. The bottom line of the summary shows the percentage of sampled documents over all available documents.
```

> When *Lens* presents the top 10 values distribution every other value, together with documents with no data for the inspected field, are shown with the "Other" entry in the distribution chart.

Reviewer: I don't think "no data" is counted in value_count, did you see a case where we would include it?

```suggestion
When *Lens* presents the top 10 values distribution, it also shows the percentage of "Other" values. For array value fields, the percentage distribution is considers each value in the array separate.
```

[role="screenshot"]
image::images/lens_data_info.png[Data summary window]

> Sometimes documents have no data for the field inspected: *Lens* will show a "Other" entry to represent these documents in the distribution chart.

Reviewer 3: ```suggestion
Sometimes documents have no data for the field inspected:. *Lens* shows an *Other* entry to represent these documents in the distribution chart.
```

Reviewer 3: Can you add a screenshot that shows this?

Author: I made the screenshot for it, but then dropped as I thought it was too much. Will add it back! 📸
